lib/frame_window_filtering/hog_labeling.cpp

```diff
@@ -30,6 +30,34 @@ std::vector<cv::Point> slidingWindows(int imgWidth, int imgHeight, int stepSize,
   return result;
 } // namespace MouseTrack
 
+
+int HogLabeling::slidingWindowWidth() const {
```

Comment:
Thanks for updating these, but what about block size/stride? I would suggest either parameterizing them or at least defining them based on the window parameters.

Also, setters/getters are good, but why not just initialize everything in the constructor? Would these parameters be modified dynamically?

Reply:
As I understand it, the patch from which we extract the hog feature vector always has the same size as the sliding window? Or do you mean the cells of a patch? Until now, patchSize=slidingWindowSize and stride were the only values we've been tweaking, if it shows useful to change other things, I'll add accessors later.

Setter/Getter vs. Constructor: 

Probably mostly a design choice, I've never been taught anything official about this. The constructor has the responsibility to create a valid object. This implies, that all information needed to create a valid object, has to be passed to the constructor. 
One design rule I try to follow, is to always put "reasonable" default values if available. Hence when there's a good default value, I don't see a point in putting it in the constructor and instead make it available as property one can change.

As always, I think there are two sides: It is probably slightly more efficient to pass stuff to the constructor as you only have to write the value once. On the other hand side I think it is more important to have a flexible and robust design. Experience taught me, that complicated constructors can be very annoying, for example when you write a factory for the class.
As this module's constructor is, by design, called "rarely", I prefer robust design over efficiency.